### PR TITLE
Dynamic dates, issue #258

### PR DESCRIPTION
**Pull request by IlinDmitry**
Hello.
I'm practicing Rails. My first PR in this project.

### CODE REVIEW

File: app/controllers/site_controller.rb [hunk 1]
```diff
@@ -3,6 +3,50 @@ class SiteController < ApplicationController
   respond_to :html
 
   def index
+    a_shed = AnnualSchedule.current
+    @shedule = []
```
Comment (marlabrizel) [inline app/controllers/site_controller.rb]: Can you please correct the spelling here? `@shedule` -> `@schedule`

File: app/controllers/site_controller.rb [hunk 2]
```diff
@@ -3,6 +3,50 @@ class SiteController < ApplicationController
   respond_to :html
 
   def index
+    a_shed = AnnualSchedule.current
```
Comment (marlabrizel) [inline app/controllers/site_controller.rb]: Hi! We like to use full-name variables here to improve readability. Would you mind renaming this variable to something more descriptive? I would suggest `current_schedule`.

File: app/controllers/site_controller.rb [hunk 3]
```diff
@@ -3,6 +3,50 @@ class SiteController < ApplicationController
...
+    @shedule = []
+
+    unless a_shed.nil?
```
Comment (marlabrizel) [inline app/controllers/site_controller.rb]: This is probably too much logic to have in the controller. Can you please push this down into the appropriate model? You will probably want to break it up into multiple, composable methods. You may also wish to consider refactoring this logic away from the heavy use of conditionals.

File: app/controllers/site_controller.rb [hunk 4]
```diff
@@ -34,4 +78,14 @@ def page_partial
     template_name = "site/#{params[:page].presence || 'index'}".underscore
     lookup_context.find(template_name).virtual_path
   end
+
+  def start_date(date_start)
```
Comment (marlabrizel) [inline app/controllers/site_controller.rb]: These methods also probably don't belong in the controller.

File: app/views/site/program.html.slim [hunk 5]
```diff
@@ -44,9 +44,7 @@ section.common
   .section-content.sm-centered
     h1 Key Dates for #{AnnualSchedule.current.year}
     h6 Kickoff Event: March 21st, 2018
-    h6 Call for Proposals: April 4th - 30th, 2018
-    h6 Voting on Sessions: May 2nd to 18th,  2018
-    h6 Events Announced and Registration Open: July 2018
-    h6 Denver Startup Week: September 24th – 28th
+    - @shedule.each do |sh|
```
Comment (marlabrizel) [inline app/views/site/program.html.slim]: can you please fix the spelling and variable naming here? This should probably be something like `@current_schedule.each do |milestone|`

File: app/views/site/program/selection.html.slim [hunk 6]
```diff
@@ -8,12 +8,10 @@ section.common.sm-centered
...
 section.common.sm-centered
   .section-content
-    h1 Key Dates for 2018
+    h1 Key Dates for #{AnnualSchedule.current.year}
     h6 2018 Kickoff Event: March 21st, 2018
-    h6 Call for Proposals: April 4th - 30th, 2018
-    h6 Voting on Sessions: May 2nd to 18th,  2018
-    h6 Events Announced and Registration Open: July 2018
-    h6 Denver Startup Week: September 24th – 28th
+    - @shedule.each do |sh|
```
Comment (marlabrizel) [inline app/views/site/program/selection.html.slim]: can you please fix the spelling and variable naming here? This should probably be something like @current_schedule.each do |milestone|

**Comment by jayzes** [comment]
Hi @IlinDmitry, thanks for the PR!

I think we may be able to accomplish this with a bit less code using a pattern like the one in this view: https://github.com/denverstartupweek/dsw-site/blob/fe99ebff9b3907db7ff55f92c95b66a13ee18bd4/app/views/layouts/shared/faq/_timeline.html.slim#L10

That way we don't have to assemble the date string on every request to a static page (even ones that don't necessarily render that data) and we can reuse the existing helpers and formats.

**Comment by IlinDmitry** [comment]
Hello!
Thank you all for review.
@jayzes, you mean like this? Should i also add tests for it? Because  i didn't see any tests for `/home/dima/RubymineProjects/dsw-site/app/views/site/get_involved/faq.html.slim`

**Comment by jayzes** [comment]
@IlinDmitry this looks great! I like that you pulled that repeated content into a partial.

On the test front, we've built a few smoke tests that help provide some quick verification that these content-oriented pages are syntactically correct and render properly. You can see those at https://github.com/denverstartupweek/dsw-site/blob/master/spec/features/content_pages_spec.rb. They're not amazingly great tests, but they do help catch egregious errors, so there's some benefit. It looks like the program page is already covered by that existing suite (including your new partial), so I think we're good to go there.

**Comment by jayzes** [comment]
@marlabrizel do you agree?

**Pull request merged**